Autopilot/PathManager/Inc/landingTakeoffManager.hpp

```diff
@@ -0,0 +1,167 @@
+        */
+        static _LandingPath createSlopeWaypoints(const Telemetry_PIGO_t * input, const float * currentAltitude);
+
+        /*
+            This function translates waypoint output data into landing takeoff output data
+            Takes in a waypoint data out structure, reformats it, and outputs a landingTakeoffOutput instance
+
+            @param _WaypointManager_Data_Out * outputData - this variable will hold the output data that is outputted by waypointManager functions
+
+            @return _LandingTakeoffOutput - this function will output a landing and takeoff output struct
+        */
+        static _LandingTakeoffOutput translateWaypointCommands(const _WaypointManager_Data_Out * outputData);
+
+        /*
+            This function translates landing takeoff data to output data
+            Takes in a landingTakeoffOutput instance, translates its information into the 2 types of coordinated turn inputs
+
+            @param _LandingTakeoffOutput* outputData - this pointer points to the output of the landing and takeoff states
+            @param SFOutput_t* sensorOutput - this pointer points to the sensor output of sensorFusion
+            @param IMUOutput_t* imuOutput - this pointer points to the imu output of sensorFusion
+            @param CoordinatedTurnInput_t * turnInput - this pointer points to the coorinated turn input structure that needs to be changed
+            @param AltitudeAirspeedInput_t * altitudeAirspeedInput - this pointer points to the altitude and airspeed input structure that needs to be changed
+        */
+        static void translateLTSFCommandsToCoordTurns(const _LandingTakeoffOutput* outputData, const SFOutput_t* sensorOutput, const IMU_Data_t* imuOutput, CoordinatedTurnInput_t* turnInput, AltitudeAirspeedInput_t* altitudeAirspeedInput);
+};
+
+/************************
+    TAKEOFF SECTION
+************************/
+
+constexpr static double EXIT_TAKEOFF_ALTITUDE {50}; //in meters, difference in altitude
+constexpr static double ROTATION_SPEED_NO_PACKAGE {4}; //in meters per second
+constexpr static double ROTATION_SPEED_WITH_PACKAGE {6}; //in meters per second
+constexpr static double CLIMB_SPEED_NO_PACKAGE {5}; // in meters per second
+constexpr static double CLIMB_SPEED_WITH_PACKAGE {7}; // in meters per second
+constexpr static double DISTANCE_OF_TAKEOFF {100}; // in meters, just used to approximate the horizontal location
+
+class TakeoffManager{
+
+    public:
+        /*
+            This function returns the desired rotation speed limit, where the plane will transition from roll to climb
+            
+            @param bool * ifPackage - this boolean holds if there is a package on board the aircraft
+
+            @return double - this function will return the desired rotation speed for the aircraft
+        */
+        static double desiredRotationSpeed(const bool * ifPackage);
```

Comment:
As a general rule. Individual built in variables, like bools, ints, and doubles, should be passed by value. Structs should be passed by reference.

Reply:
oh yeah ok, that makes sense. Will fix